File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
       if(x<0){
        return false;
    }
    
    int original=x;
    long long reversed=0;
    
    while(x>0){
        int digits=x%10;
        reversed=reversed*10+digits;
        x=x/10;
    }
    return original==reversed;
    }
};

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp=x;
        int sum=0;
        if (x<0) return false;
        while(temp!=0){
            sum=sum*10+(temp%10);
            temp=temp/10;
        }
        return x==sum;
    }
}